### PR TITLE
use typechain hardhat plugin

### PR DESCRIPTION
**Pull request by ItsNickBarry**
@Lumyo This branch includes a mostly-working installation of the Typechain Hardhat plugin.  However, after running `hh clean`, `hh test` fails the first time unless `hh compile` has been run explicitly.  It works as expected on subsequent calls.  Is this the issue that you mentioned?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,6 +11,4 @@ yarn.lock
 
 .env
 
-dist
```
Comment (Lumyo) [inline .gitignore:14]: You need to leave `dist` in the `.gitignore`, as this folder is generated by the typescript compiler and should not be commited

**Comment by Lumyo** [comment]
I had some other error during compilation, but I dont have it anymore. 
So it was probably related to something else

File: .gitignore (hunk 1 above)
Comment (ItsNickBarry) [reply .gitignore:14]: That's what I thought, but I don't see `dist` anywhere.  When is it generated?

Comment (Lumyo) [reply .gitignore:14]: At the root of the project when you run the typescript compiler through `tsc`

Comment (ItsNickBarry) [reply .gitignore:14]: When is `tsc` run?

Comment (Lumyo) [reply .gitignore:14]: You can run it anytime when you wanna check that your code compiles without error (some errors might not be displayed by the IDE, but will be caught by the TS compiler) or when you wanna transpile your code to JS for deployment / publishing

Comment (ItsNickBarry) [reply .gitignore:14]: Okay, fixed.

**Pull request merged**